### ISSUE TITLE
Split by intent with binary answers classifier buggy interface

### ISSUE THREAD

**Issue by paulobernardoaf** [nyaruka/floweditor#890]
Hello! There are a few issues with the `split by intent` card when `binary answers` is selected as a classifier that is synchronized from BotHub. 

**First one:**
- As you can see in this image: 
<img src="https://user-images.githubusercontent.com/30026625/88331821-f65b5c80-cd03-11ea-947f-d9a1ff788da3.png" width="475" height="400">

- Both `affirmative` and `negative` are selected.

- After saving it with the `Ok` button and then opening the card again to edit, the selected intent is not shown:

<img src="https://user-images.githubusercontent.com/30026625/88332537-1a6b6d80-cd05-11ea-800a-82bdd9d90244.png" width="475" height="400">

**Second one**

- When setting the confidence rate the intent field shows a warning: `Intent is required ` and this warning can only be fixed by manually setting the intent to the same value as before.

There's a gif to show the issue:

![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/30026625/88341316-c87e1400-cd13-11ea-8399-aa6e53501017.gif)

**Comment by matmsa27** [comment]
Hi everyone, it is reproducible on your side?? There is a way to us open a pr to fix this issue?

**Comment by paulobernardoaf** [comment]
Hello everyone, I've looked at the code and discovered that:
- The `intents` list created [here](https://github.com/nyaruka/floweditor/blob/master/src/components/flow/routers/case/CaseElement.tsx#L349) is correct but when the `CaseElement` component is loaded, that `intents` array is empty and it is passed to the `SelectElement` component [here](https://github.com/nyaruka/floweditor/blob/master/src/components/flow/routers/case/CaseElement.tsx#L370), causing the first render to not display the list on the selector. The expected behavior is that when the `intents` array is updated, the `SelectElement` component will also update but that's not happening. 
- I've been able to show the correct select but in exchange, there's a delay on the component render and it's not recommended for the end-user experience. Example of how to do that:

```js
{intents.length &&
<SelectElement ... >
...
</SelectElement>
}
```
